kani-compiler/src/codegen_cprover_gotoc/overrides/loop_contracts_hooks.rs

```diff
@@ -0,0 +1,37 @@
+// Copyright Kani Contributors
+// SPDX-License-Identifier: Apache-2.0 OR MIT
+
+use super::hooks::GotocHook;
+use crate::codegen_cprover_gotoc::codegen::bb_label;
+use crate::codegen_cprover_gotoc::GotocCtx;
+use crate::kani_middle::attributes::KaniAttributes;
+use cbmc::goto_program::{CIntType, Expr, Stmt, Type};
+use rustc_middle::ty::TyCtxt;
+use rustc_span::Symbol;
+use stable_mir::mir::mono::Instance;
+use stable_mir::mir::{BasicBlockIdx, Place};
+use stable_mir::ty::Span;
+
+pub struct LoopInvariantRegister;
+
+impl GotocHook for LoopInvariantRegister {
```

Comment:
tbh, I would just keep it in the same file as the other hooks. But not a big deal, can you document what this is expanding to? Thanks!